docs/guides/tutorials/rest_apis_with_fastapi.rst

```diff
@@ -1167,16 +1170,403 @@ payload:
+        return response
+
+The sign-up endpoint sends a POST request to the EdgeDB Auth server to register
+a new user. It also generates a PKCE verifier and sets it as an HttpOnly cookie
+in the response.
+
+**Sign-in endpoint**
+
+.. code-block:: python
+    :caption: app/auth.py
+
+    @router.post("/auth/signin")
+    async def handle_signin(request: Request):
+        body = await request.json()
+        email = body.get("email")
+        password = body.get("password")
+        provider = body.get("provider")
+
+        if not email or not password or not provider:
+            raise HTTPException(status_code=400, detail="Missing email, password, or provider.")
+
+        verifier, challenge = generate_pkce()
+        authenticate_url = f"{EDGEDB_AUTH_BASE_URL}/authenticate"
+        response = httpx.post(authenticate_url, json={
+            "challenge": challenge,
+            "email": email,
+            "password": password,
+            "provider": provider,
+        })
+
+        if response.status_code != 200:
+            return JSONResponse(status_code=400, content={"message": "Authentication failed"})
+
+        code = response.json().get("code")
+        token_url = f"{EDGEDB_AUTH_BASE_URL}/token"
+        token_response = httpx.get(token_url, params={"code": code, "verifier": verifier})
+
+        if token_response.status_code != 200:
+            return JSONResponse(status_code=400, content={"message": "Token exchange failed"})
+
+        auth_token = token_response.json().get("auth_token")
+        response = JSONResponse(content={"message": "Authentication successful"})
+        response.set_cookie(key="edgedb-auth-token", value=auth_token, httponly=True, secure=True, samesite='strict')
+        return response
+
+The sign-in endpoint sends a POST request to the EdgeDB Auth server to authenticate
+a user. It then retrieves the code from the response and exchanges it for an auth 
+token. The token is set as an HttpOnly cookie in the response.
+
+**Add the auth endpoints to the FastAPI application**
+
+Finally, add the auth endpoints to the FastAPI application:
+
+.. code-block:: python-diff
+    :caption: app/main.py
+
+    + fast_api.include_router(events.router)
+
+Creating a new user in the sign-up endpoint
+-------------------------------------------
+
+Now, let's automatically create a new user in the database when a user signs up.
+We'll use the ``create_user_async_edgeql`` query we generated earlier 
+to achieve this, but we'll need to modify it slightly to link it to the
+EdgeDB Auth identity.
+
+First, let's update the EdgeDB schema to include a new field in the User type
+to store the EdgeDB Auth identity and a new ``current_user`` type.
+
+.. code-block:: sdl-diff
+    :caption: dbschema/default.esdl
+    
+    + global current_user := assert_single(
+    +     ((
+    +         select User { id, name }
```

Comment:
Minor nit: Remove this `{ id, name }` shape from the select: the shape is used when selecting from the global, not in the global definition itself.